src/Form/LegacyConsumer/Renderer.php

```diff
@@ -0,0 +1,345 @@
+<?php
+
+namespace Give\Form\LegacyConsumer;
+
+use DOMDocument;
+use Give\Framework\FieldsAPI\FieldCollection;
+use Give\Framework\FieldsAPI\FormField\FieldTypes;
+
+class Renderer {
+	public function __construct( FieldCollection $fieldCollection ) {
+		$this->document        = new DOMDocument( '1.0', 'utf-8' );
+		$this->fieldCollection = $fieldCollection;
+		$this->currentRoot     = $this->document;
+	}
+
+	/**
+	 * Render the field collection
+	 */
+	public function render() {
+		$this->fieldCollection->walk(
+			function ( $field ) {
+				// Determine the needs of the field based on the type
+				$config = static::deriveConfigFromType( $field->getType() );
+
+				if ( $field->getType() === FieldTypes::TYPE_SECTION ) {
```

Comment:
I was thinking that instead of a "section" type that it could be a `GroupNode` in the field collection, but the `WalkNodes::walkCollection` method "walks past `GroupNode`s," so unless that implementation was changed, "section" or "fieldset" needs to remain a field type.